### ISSUE TITLE
No miner folder, no Bin folder

### ISSUE THREAD

**Issue by asdfg323** [MultiPoolMiner/MultiPoolMiner#2235]
I downloaded the newest version and was unable to successfully start it. Kept getting this warning: "WARNING: No miners available. "

I noticed that there was no Bin folder or Miners folder like previous releases. (Just a MinersLegacy folder)

I downloaded a fresh excavator and made a new Bin folder to put it in per instructions.

Please advise, thanks.

**Comment by asdfg323** [comment]
Also, tried copying both of those folders (even though instructions say not to) from a previous version (that is running right now) and still got the same error.

**Comment by UselessGuru** [comment]
Bin & Download folder will be created automatically when the miners get downloaded.

You do no longer need to download excavator, MPM will handle all the downloads.

If you need more assistance then please save the output of 'http://localhost:3999/runningconfig.html' to a file and upload it here.

**Comment by asdfg323** [comment]
[ouput.txt](https://github.com/MultiPoolMiner/MultiPoolMiner/files/2569130/ouput.txt)


Here you go. Thanks for taking a look. Only things I did from new install is change the start.bat commands with wallet info, ect, deleted all pools except mph, deleted cast.txt in minerslegacy folder, and added excavator to created bin folder.

**Comment by asdfg323** [comment]
![capture](https://user-images.githubusercontent.com/44918692/48310081-1e108e00-e53c-11e8-8055-2bef639fc401.JPG)
After running 5 min or so, heres what I see in the powershell consoles.

**Comment by UselessGuru** [comment]
> Here you go. Thanks for taking a look. Only things I did from new install is change the start.bat commands with wallet info, ect, deleted all pools except mph, deleted cast.txt in minerslegacy folder, and added excavator to created bin folder.

You write 

>deleted cast.txt

 This means you did not install to a new directory. This might be the reason for your problem.

Here is what I tested with 3.1.0 Beta 10 (**install to a new directory**):
I copied your config as confix.txt to the MPM folder and started MPM with 'Start.cmd'.

It started downloading and then mining as designed. So this makes it very hard to aid with troubleshooting. 

A hint: If you are starting with a new installation I recommend that you enable 'CreateMinerInstancePerDeviceModel and 'UseDeviceNameForStatsFileNaming'.
Note :Neither of these two settings have in impact on your issue.

**Comment by asdfg323** [comment]
I downloaded a fresh copy of 3.0.1a, it comes with cast.exe, which I deleted. The only step I took before the steps I stated I took, was taking the .zip file from github, and opening it into a new folder.

Where do I enable the create and use settings you wrote about?
Here is my start.bat


![capture](https://user-images.githubusercontent.com/44918692/48317523-6cf40d00-e5a8-11e8-8d96-5bd5a799b7a4.JPG)

**Comment by asdfg323** [comment]
From my understanding cast.exe is a vega file, I only have Nvidia cards on this PC, and it was causing problems in the past, so long ago I wrote my "upgrade guide" between versions of MPM. I am in the process of upgrading from 2.7.2.7, and luckily skipped the hacked version.

Here is the entirety of up upgrade guide lol.

![capture](https://user-images.githubusercontent.com/44918692/48317633-b09b4680-e5a9-11e8-9a27-4c3827050b0b.JPG)

**Comment by UselessGuru** [comment]
@asdfg323 

My Upgrade guide:

- Download the latest Beta from here: https://github.com/MultiPoolMiner/MultiPoolMiner/releases/tag/3.1.0
- Unpack to a new empty folder
- replace wallet and username in the start batch file(s)
- add '-CreateMinerInstancePerDeviceModel and '-UseDeviceNameForStatsFileNaming' to  the start batch file(s) - optional
- Start MPM

There is no manual downloading or deleting of files necessary

**Comment by UselessGuru** [comment]
What driver version are you using? Did you install new NVIDIA graphics drivers?
I am just encountering a similar issue where MPM cannt detect my NVIDIA card because of a corrupt OpenCL installation (which I am just trying to fix).

**Comment by paulpoco** [comment]
Every now and then use this https://www.guru3d.com/files-details/display-driver-uninstaller-download.html

**Comment by asdfg323** [comment]
http://localhost:3999/devices gives me a blank white page with 2 brackets separated by a line (Below)


[

]

**Comment by UselessGuru** [comment]
> gives me a blank white page with 2 brackets separated by a line (Below)

Appently MPM cannot detect our graphics cards (something is wrong with the OpenGL library in your rig). Theses are part of your graphics card driver.

Furthermore your config file is not formatted properly, but that is not the problem.

DeviceName schould be an array and not a string, e.g.:

```
    "DeviceName":  [
                       "AMD",
                       "CPU",
                       "NVIDIA"
                   ],
```

Can you see if CPU mining would start? That would prove that the OpenGL libraries are not properly installed on your computer.

**Comment by asdfg323** [comment]
Added CPU to the batch file, and it started working with  http://localhost:3999/devices populating. Removed CPU again and same problem.

In a couple days I am rebuilding my computer anyways, so I'll start with a completely fresh install of everything, again lol.

Last time I only installed the graphics driver, I never install 3D or audio drivers. Before my rebuild, Ill try to nuke my graphics drivers then fresh install again to troubleshoot. Then if that doesnt work, do a full install and manually remove the 3D and audio driver bloatware. Then at least you'll have an answer and this might show up in google for others.
